src/types/dense.jl

```diff
@@ -3,14 +3,18 @@ using Base: @propagate_inbounds
 struct DenseTaggedImage{T, N, O <: Unsigned,AA <: AbstractArray} <: AbstractDenseTIFF{T, N}
     data::AA
     ifds::Vector{IFD{O}}
+    pagecache::Matrix{UInt8}
```

Comment:
I'm not a fan of placing the write cache in the struct itself. I think it would be sufficient to construct this in the `write` function below.